### ISSUE TITLE
Replace TabActivity with AppCompatActivity

### ISSUE THREAD

**Issue by 12people** [jonasoreland/runnerup#648]
TabActivity has long been deprecated, and its use prevents niceties from the support library—vector drawables, for example.

Please use AppCompatActivity for MainLayout and fragments for its individual components (StartActivity, FeedActivity, HistoryActivity, SettingsActivity).

All of this is supported on Froyo, so there should be no compatibility issues.

**Comment by gerhardol** [comment]
A partial conversion to AppCompat is available in branch: https://github.com/gerhardol/runnerup/tree/feature/appcompat-changes

**Comment by gerhardol** [comment]
Migration to androidx and AppCompatActivity done where possible. Only TabActivity and SettingsActivity remains, which is quite some changes.

**Comment by roberi** [comment]
> Only TabActivity and SettingsActivity remains, which is quite some changes.

I have been working on migrating all settings to AndroidX Preferences. Before I make a PR I wonder if there is some special reason why the SettingsActivity has no Toolbar (app bar)? Both StartActivity and HistoryActivity have one.

![image](https://github.com/jonasoreland/runnerup/assets/10847866/d9f38518-3a73-42f7-89fe-3bdc2e3da817)

Should we add a Toolbar to SettingsActivity as well?

**Comment by gerhardol** [comment]
> > Only TabActivity and SettingsActivity remains, which is quite some changes.
> 
> I have been working on migrating all settings to AndroidX Preferences. Before I make a PR I wonder if there is some special reason why the SettingsActivity has no Toolbar (app bar)? Both StartActivity and HistoryActivity have one.

> Should we add a Toolbar to SettingsActivity as well?

No specific reason, it just happen to be this way....

The recommended handling for settings is not to have it in the toolbar. But with only two other tabs, it is not a space issue.

Looking forward for this - I have gotten stuck every time I started.

**Comment by roberi** [comment]
> The recommended handling for settings is not to have it in the toolbar.

I'm not sure I understand. Or perhaps we misunderstand each other? Why not like in the right picture?

![image](https://github.com/jonasoreland/runnerup/assets/10847866/1d7e4fed-e80a-47ac-b500-da8353d5d049)

> Looking forward for this - I have gotten stuck every time I started.

Looking forward for your feedback. =)

**Comment by gerhardol** [comment]
> > The recommended handling for settings is not to have it in the toolbar.
> 
> I'm not sure I understand. Or perhaps we misunderstand each other? Why not like in the right picture?

My comment was related to Settings as a tab is not recommended.

Add the app bar...

**Comment by roberi** [comment]
@gerhardol, have you or anyone else started any work to replace TabActivity? If not, I can give it a try.

**Comment by gerhardol** [comment]
> @gerhardol, have you or anyone else started any work to replace TabActivity? If not, I can give it a try.

No, I have not tried that. I am not aware that anyone else is working on that.
It would be welcome!

**Comment by gerhardol** [comment]
@roberi If you see any inconsistencies and improvements, just fix them. The GUI could be improved...
